File: oauth2-authorization-server-extended/authorization-server/src/main/java/com/onlyteo/sandbox/validator/ValidPassword.java
package com.onlyteo.sandbox.validator;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.*;

@Constraint(validatedBy = PasswordConstraintValidator.class)
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface ValidPassword {

    String message() default "{validation.password.Valid.message}";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

    String field() default "password";

    String confirmField() default "confirmPassword";
}

File: spring-boot-oauth2-extended-authorization-server/authorization-server/src/main/kotlin/com/onlyteo/sandbox/validator/ValidPassword.kt
package com.onlyteo.sandbox.validator

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Constraint(validatedBy = [PasswordConstraintValidator::class])
@Target(AnnotationTarget.TYPE, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class ValidPassword(
    val message: String = "{validation.password.Valid.message}",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
    val field: String = "password",
    val confirmField: String = "confirmPassword",
)
